qcodes/instrument_drivers/QuantumDesign/DynaCoolPPMS/DynaCool.py

```diff
@@ -0,0 +1,273 @@
+                           label='Temperature',
+                           unit='K',
+                           get_parser=partial(DynaCool._pick_one, 1, float),
+                           get_cmd='TEMP?')
+
+        self.add_parameter('temperature_setpoint',
+                           label='Temperature setpoint',
+                           unit='K',
+                           vals=vals.Numbers(1.8, 400),
+                           set_cmd=partial(self._temp_setter,
+                                           'temperature_setpoint'),
+                           get_cmd=partial(self._temp_getter,
+                                           'temperature_setpoint'))
+
+        self.add_parameter('temperature_rate',
+                           label='Temperature settle rate',
+                           unit='K/s',
+                           vals=vals.Numbers(0.0002, 0.3),
+                           set_parser=lambda x: x*60,  # conversion to K/min
+                           get_parser=lambda x: x/60,  # conversion to K/s
+                           set_cmd=partial(self._temp_setter,
+                                           'temperature_rate'),
+                           get_cmd=partial(self._temp_getter,
+                                           'temperature_rate'))
+
+        self.add_parameter('temperature_settling',
+                           label='Temperature settling mode',
+                           val_mapping={'fast settle': 0,
+                                        'no overshoot': 1},
+                           set_cmd=partial(self._temp_setter,
+                                           'temperature_settling'),
+                           get_cmd=partial(self._temp_getter,
+                                           'temperature_settling'))
+
+        self.add_parameter('temperature_state',
+                           label='Temperature tracking state',
+                           val_mapping={"tracking": 2,
+                                        'stable': 1,
+                                        'near': 5,
+                                        'chasing': 6,
+                                        'pot operation': 7,
+                                        'standby': 10,
+                                        'diagnostic': 13,
+                                        'impedance control error': 14,
+                                        'failure': 15},
+                           get_parser=partial(DynaCool._pick_one, 2, int),
+                           get_cmd='TEMP?')
+
+        self.add_parameter('field',
+                           label='Field strength',
+                           unit='A/m',
+                           get_cmd=self._present_field_getter)
+
+        self.add_parameter('field_setpoint',
+                           label='Field setpoint',
+                           unit='A/m',
+                           get_parser=lambda x: x/79.57747,  # Oe to A/m
```

Comment:
shall this conversion be extracted? I saw it is also used in `_present_field_getter`